### ISSUE TITLE
Unsupported type function 

### ISSUE THREAD

**Issue by wszarmach-koia** [gruntwork-io/terragrunt#2033]
While using [type](https://www.terraform.io/language/functions/type) function which is available since Terraform 1.0 I get the following messages:

```
ERRO[0000] Error: Call to unknown function
    
ERRO[0000]   on /home/test/project/terragrunt/terragrunt.hcl line 20, in locals: 
ERRO[0000]   20:   vars           = { for key, value in local.vars: key => value if type(value) == "string" } 
ERRO[0000]                                              
ERRO[0000] There is no function named "type".
          
ERRO[0000] Encountered error while evaluating locals.   
ERRO[0000] /home/test/project/terragrunt/terragrunt.hcl:20,72-76: Call to unknown function; There is no function named "type". 
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```

Terragrunt versions: 0.35.13 & 0.36.3
Terraform 1.1.7
OS: Linux, x86_64

The function is fine itself. Verified with **terragrunt console** for the same project.

**Comment by yorinasub17** [comment]
Unfortunately, we can't import any terraform function past v0.15.3 due to a change in the internal structure of terraform that made all the function definitions private to the project.

I thought we documented this in the docs, but I do see that we haven't done so, so this is a documentation bug.

**Comment by wszarmach-koia** [comment]
Thanks for the answer. I fully understand it. Is there any chance that it would be implemented inside Terragrunt itself ? This is one of those functions that cannot be easily replaced by some code.

**Comment by wszarmach-koia** [comment]
Right, I didn't notice that part.I believe that we can close the ticket under given circumstances. Though, it would be useful. Thanks for your your time.

**Comment by jakubigla** [comment]
So how one can recognise if value is a map or not? Needed for deep merge workaround on terragrunt level
